node/core/approval-voting/src/old_tests.rs

```diff
@@ -0,0 +1,1964 @@
+				cert: garbage_assignment_cert(
+					AssignmentCertKind::RelayVRFModulo { sample: 0 }
+				),
+				tranche: maximum_broadcast,
+				validator_index: ValidatorIndex(4),
+				triggered: false,
+			}),
+			our_approval_sig: None,
+			assignments: bitvec::bitvec![BitOrderLsb0, u8; 0; 4],
+			approved: false,
+		};
+
+		approval_db::v1::CandidateEntry {
+			candidate: Default::default(),
+			session: 1,
+			block_assignments: vec![(block_hash, approval_entry)].into_iter().collect(),
+			approvals: bitvec::bitvec![BitOrderLsb0, u8; 0; 4],
+		}.into()
+	};
+
+	let tranche_now = maximum_broadcast;
+	assert!(should_trigger_assignment(
+		candidate_entry.approval_entry(&block_hash).unwrap(),
+		&candidate_entry,
+		RequiredTranches::Pending {
+			maximum_broadcast,
+			clock_drift: 0,
+			considered: 10,
+			next_no_show: None,
+		},
+		tranche_now,
+	));
+}
+
+#[test]
+fn assignment_not_triggered_if_at_maximum_but_clock_is_before() {
+	let block_hash = Hash::repeat_byte(0x01);
+	let maximum_broadcast = 10;
+
+	let candidate_entry: CandidateEntry = {
+		let approval_entry = approval_db::v1::ApprovalEntry {
+			tranches: Vec::new(),
+			backing_group: GroupIndex(0),
+			our_assignment: Some(approval_db::v1::OurAssignment {
+				cert: garbage_assignment_cert(
+					AssignmentCertKind::RelayVRFModulo { sample: 0 }
+				),
+				tranche: maximum_broadcast,
+				validator_index: ValidatorIndex(4),
+				triggered: false,
+			}),
+			our_approval_sig: None,
+			assignments: bitvec::bitvec![BitOrderLsb0, u8; 0; 4],
+			approved: false,
+		};
+
+		approval_db::v1::CandidateEntry {
+			candidate: Default::default(),
+			session: 1,
+			block_assignments: vec![(block_hash, approval_entry)].into_iter().collect(),
+			approvals: bitvec::bitvec![BitOrderLsb0, u8; 0; 4],
+		}.into()
+	};
+
+	let tranche_now = 9;
+	assert!(!should_trigger_assignment(
+		candidate_entry.approval_entry(&block_hash).unwrap(),
+		&candidate_entry,
+		RequiredTranches::Pending {
+			maximum_broadcast,
+			clock_drift: 0,
+			considered: 10,
+			next_no_show: None,
+		},
+		tranche_now,
+	));
+}
+
+#[test]
+fn assignment_not_triggered_if_at_maximum_but_clock_is_before_with_drift() {
+	let block_hash = Hash::repeat_byte(0x01);
+	let maximum_broadcast = 10;
+
+	let candidate_entry: CandidateEntry = {
+		let approval_entry = approval_db::v1::ApprovalEntry {
+			tranches: Vec::new(),
+			backing_group: GroupIndex(0),
+			our_assignment: Some(approval_db::v1::OurAssignment {
+				cert: garbage_assignment_cert(
+					AssignmentCertKind::RelayVRFModulo { sample: 0 }
+				),
+				tranche: maximum_broadcast,
+				validator_index: ValidatorIndex(4),
+				triggered: false,
+			}),
+			our_approval_sig: None,
+			assignments: bitvec::bitvec![BitOrderLsb0, u8; 0; 4],
+			approved: false,
+		};
+
+		approval_db::v1::CandidateEntry {
+			candidate: Default::default(),
+			session: 1,
+			block_assignments: vec![(block_hash, approval_entry)].into_iter().collect(),
```

Comment:
Seems to be rather repetitive and could be parameterize by a few variables - something worth to consider.

Reply:
Awesome feedback... exactly what I needed! Thank you!